wagtail/core/models.py

```diff
@@ -563,39 +563,55 @@ def _update_descendant_url_paths(self, old_url_path, new_url_path):
                 Value(new_url_path),
                 Substr('url_path', len(old_url_path) + 1))))
 
-    #: Return this page in its most specific subclassed form.
+    #: versionadded:: 2.10
+    #: Return this page's ``content_type`` value from the
+    #: model's cached manager, avoiding a database query.
     @cached_property
-    def specific(self):
+    def content_type_cached(self):
         """
-        Return this page in its most specific subclassed form.
+        Return this page's ``content_type`` value from the
+        model's cached manager, avoiding a database query.
```

Comment:
I've updated the docstring to be a little clearer on this. There will be a database query if the ContentType isn't already in-memory. Django takes lots of opportunities to update the cache when certain things happen, but I don't think it ever triggers a 'full refresh' at any point, so it can be a bit hit-and-miss as to whether the database is queried. It's a common gotcha in `assertNumQueries()` tests, as the result can vary depending on whether another test has requested the content type in question.